tests/phpunit/agentic-commerce/class-wc-stripe-agentic-commerce-product-exclusion-test.php

```
@@ -0,0 +1,136 @@
+		$product->delete( true );
+	}
+
+	/**
+	 * A non-WC_Product payload preserves the incoming vote instead of fataling.
+	 */
+	public function test_filter_callback_tolerates_non_product_payload(): void {
+		$exclusion = new WC_Stripe_Agentic_Commerce_Product_Exclusion();
+
+		$this->assertTrue( $exclusion->filter_should_sync_product( true, null ) );
+		$this->assertFalse( $exclusion->filter_should_sync_product( false, null ) );
+	}
```

PHPMD (2.15.0)

[error] The method test_filter_callback_tolerates_non_product_payload is not named in camelCase. (undefined)

(CamelCaseMethodName)